community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/executionplan/UpdateCounter.scala

```diff
@@ -20,6 +20,12 @@
 package org.neo4j.cypher.internal.compiler.v2_3.executionplan
 
 class UpdateCounter {
+  def offsetForHeaders(): Unit = {
+    if (uncommittedRows != 0)
+      throw new IllegalStateException("Header offset must be accounted for at the beginning")
+    uncommittedRows = -1
+  }
```

Comment:
This is how we ensure that header rows are not counted the same way as data rows.